firedrake/interpolation.py

```diff
@@ -610,30 +627,40 @@ def _get_tensor(self) -> op2.Mat | Function | Cofunction:
                     val = Constant(finfo.min)
                 f.assign(val)
         elif self.rank == 2:
-            Vrow = self.interpolate_args[0].function_space()
-            Vcol = self.interpolate_args[1].function_space()
-            if len(Vrow) > 1 or len(Vcol) > 1:
-                raise NotImplementedError("Interpolation matrix with MixedFunctionSpace requires MixedInterpolator")
-            Vrow_map = get_interp_node_map(self.source_mesh, self.target_mesh, Vrow)
-            Vcol_map = get_interp_node_map(self.source_mesh, self.target_mesh, Vcol)
-            sparsity = op2.Sparsity((Vrow.dof_dset, Vcol.dof_dset),
-                                    [(Vrow_map, Vcol_map, None)],  # non-mixed
-                                    name=f"{Vrow.name}_{Vcol.name}_sparsity",
-                                    nest=False,
-                                    block_sparse=True)
+            sparsity = self._get_sparsity(mat_type=mat_type)
             f = op2.Mat(sparsity)
         else:
             raise ValueError(f"Cannot interpolate an expression with {self.rank} arguments")
         return f
 
-    def _get_callable(self, tensor=None, bcs=None):
+    def _get_sparsity(self, mat_type=None) -> op2.Sparsity:
```

Comment:
```suggestion
    def _get_monolithic_sparsity(self, mat_type=None) -> op2.Sparsity:
```
Then it's clear that it can't be a mat nest. I was wondering why there was no `sub_mat_type` argument.

I also think for an internal function like this it's safer to make `mat_type` a mandatory argument. It would be easy to forget to include it and then suddenly get unexpected behaviour.